src/plugins/data/server/index_patterns/register_index_pattern_usage_collection.ts

```diff
@@ -0,0 +1,173 @@
+  indexPatternsWithRuntimeFieldCount: number;
+  scriptedFieldCount: number;
+  runtimeFieldCount: number;
+  perIndexPattern: {
+    scriptedFieldCount: CountSummary;
+    runtimeFieldCount: CountSummary;
+    scriptedFieldLineCount: CountSummary;
+    runtimeFieldLineCount: CountSummary;
+  };
+}
+
+const minMaxAvgLoC = (scripts: string[]) => {
+  const lengths = scripts.map((script) => script?.split(/\r\n|\r|\n/).length || 0).sort();
+  return {
+    min: lengths[0],
+    max: lengths[lengths.length - 1],
+    avg: lengths.reduce((col, count) => col + count, 0) / lengths.length,
+  };
+};
+
+const updateMin = (currentMin: number, newVal: number) => {
+  if (currentMin === 0 || currentMin > newVal) {
+    currentMin = newVal;
+  }
+};
+
+const updateMax = (currentMax: number, newVal: number) => {
+  if (currentMax === 0 || currentMax < newVal) {
+    currentMax = newVal;
+  }
+};
+
+export async function getIndexPatternTelemetry(indexPatterns: IndexPatternsCommonService) {
+  const ids = await indexPatterns.getIds();
+
+  const countSummaryDefaults = {
+    min: 0,
+    max: 0,
+    avg: 0,
+  };
+
+  const results = {
+    indexPatternsCount: ids.length,
+    indexPatternsWithScriptedFieldCount: 0,
+    indexPatternsWithRuntimeFieldCount: 0,
+    scriptedFieldCount: 0,
+    runtimeFieldCount: 0,
+    perIndexPattern: {
+      scriptedFieldCount: countSummaryDefaults,
+      runtimeFieldCount: countSummaryDefaults,
+      scriptedFieldLineCount: countSummaryDefaults,
+      runtimeFieldLineCount: countSummaryDefaults,
+    },
+  };
+
+  ids.forEach(async (id) => {
+    const ip = await indexPatterns.get(id);
+    // SCRIPTED FIELDS
+    const scriptedFields = ip.getScriptedFields();
+    const ipScriptedFieldLineCount = minMaxAvgLoC(scriptedFields.map((fld) => fld.script || ''));
+    results.perIndexPattern.scriptedFieldLineCount = ipScriptedFieldLineCount;
+
+    // RUNTIME FIELDS
+    const runtimeFields = ip.fields.filter((fld) => !!fld.runtimeField);
+    const runtimeFieldScripts = runtimeFields.map((fld) => fld.runtimeField?.script?.source || '');
+    const ipRuntimeFieldLineCount = minMaxAvgLoC(runtimeFieldScripts);
+    results.perIndexPattern.runtimeFieldLineCount = ipRuntimeFieldLineCount;
+
+    // index pattern count
+    if (scriptedFields.length > 0) {
+      results.indexPatternsWithScriptedFieldCount++;
+    }
+    if (runtimeFields.length > 0) {
+      results.indexPatternsWithRuntimeFieldCount++;
+    }
+    //
+
+    // field count
+    results.scriptedFieldCount += scriptedFields.length;
+    results.runtimeFieldCount += runtimeFields.length;
+    //
+
+    updateMin(results.perIndexPattern.scriptedFieldCount.min, scriptedFields.length);
+    updateMax(results.perIndexPattern.scriptedFieldCount.max, scriptedFields.length);
+
+    updateMin(results.perIndexPattern.runtimeFieldCount.min, runtimeFields.length);
+    updateMax(results.perIndexPattern.runtimeFieldCount.max, runtimeFields.length);
```

Comment:
I suggest not using mutating functions, mostly for consistency. I think something like this would be fine:
```
results.perIndexPattern.runtimeFieldCount.min = Math.min(results.perIndexPattern.scriptedFieldCount.min, scriptedFields.length)
```